httpconnect/httpconnect.go

```diff
@@ -147,28 +151,37 @@ func (f *HTTPConnectHandler) intercept(w http.ResponseWriter, req *http.Request)
 			}
 		}
 	}
-	var closeOnce sync.Once
-	go func() {
+
+	readErrCh := make(chan error)
+	op.Go(func() {
 		buf := buffers.Get()
 		defer buffers.Put(buf)
-		if _, err := io.CopyBuffer(connOut, clientConn, buf); err != nil {
-			log.Debug(err)
+		_, readErr := io.CopyBuffer(connOut, clientConn, buf)
+		if readErr != nil {
+			log.Debug(readErr)
 		}
-		closeOnce.Do(closeConns)
-
-	}()
+		readErrCh <- readErr
+	})
 	buf := buffers.Get()
 	defer buffers.Put(buf)
-	if _, err := io.CopyBuffer(clientConn, connOut, buf); err != nil {
-		log.Debug(err)
+	_, writeErr := io.CopyBuffer(clientConn, connOut, buf)
+	if writeErr != nil {
+		log.Debug(writeErr)
+	}
+
+	readErr := <-readErrCh
+	if writeErr != nil {
```

Comment:
I just realized that we're already logging these at debug level above, so the extra logging as errors was redundant.  Fixed.